calculator.c

```diff
@@ -0,0 +1,169 @@
+#include<stdio.h>
+#include<string.h>
+#define max 500
+int oprand[max], vtop = -1, otop = -1; 
+char operator[max]; 
+
+// The above global variable can be modified or accessed by any function in the program
+
+void pushvalue(int value)
+{
+	vtop = vtop + 1;
+	oprand[vtop] = value;
+}
+
+int popvalue()
+{
+    int value = oprand[vtop];
+    vtop = vtop - 1;
+    return value;
+}
+
+void pushoprator(char operator1)
+{
+	otop = otop + 1;
+	operator[otop] = operator1;
+}
+
+char popoprator()
+{
+    char operator1 = operator[otop];
+    otop = otop - 1;
+    return operator1;
+}
+
+int isDigit(char c)
+{
+	int result = 0;
+	if(c>='0' && c<='9')
+	{
+		result=1;
+	}
+	return result;
+	
+}
+
+int isoperator(char c)
+{
+	int result =0;
+	if(c=='+' || c=='-' || c=='*' || c=='/')
+	{
+		result= 1;
+	}
+	return result;
+}
+
+int precedence(char op)
+{
+	int result = 0;
+	if(op=='*' || op=='/')
+	{
+		result = 2;
+	}
+	else if(op=='+' || op=='-')
+	{
+		result = 1;
+	}
+	return result;
+}
+
+int applyoperation(int a, int b, char op)
+{
+	int answer=0;
+	switch(op)
+	{
+		case '+':
+		{
+			answer= a + b;
+			break;
+		}
+		case '-':
+		{
+			answer= a - b;
+			break;
+		}
+		case '*':
+		{
+			answer= a * b;
+			break;
+		}
+		case '/':
+		{
+			if(b == 0)
+			{
+				printf("Division by zero error!");
+				exit(1);
+			}
+			answer= a / b;
```

Comment:
these variable name are not self expressive, what does 'a' and 'b' stands here. Rename it properly